tests/suite/test_policy_ingress_class.py

```diff
@@ -0,0 +1,164 @@
+    ):
+        """
+        Test if policy with no ingress class is applied to vs
+        """
+        print(f"Create rl policy")
+        pol_name = create_policy_from_yaml(kube_apis.custom_objects, policy_src, test_namespace)
+
+        wait_before_test()
+        policy_info = read_custom_resource(kube_apis.custom_objects, test_namespace, "policies", pol_name)
+        assert (
+                policy_info["status"]
+                and policy_info["status"]["reason"] == "AddedOrUpdated"
+                and policy_info["status"]["state"] == "Valid"
+        )
+
+        print(f"Patch vs with policy: {src}")
+        patch_virtual_server_from_yaml(
+            kube_apis.custom_objects,
+            virtual_server_setup.vs_name,
+            src,
+            virtual_server_setup.namespace,
+        )
+        wait_before_test()
+
+        vs_info = read_custom_resource(kube_apis.custom_objects, virtual_server_setup.namespace, "virtualservers", virtual_server_setup.vs_name)
+        assert (
+                vs_info["status"]
+                and vs_info["status"]["reason"] == "AddedOrUpdated"
+                and vs_info["status"]["state"] == "Valid"
+        )
+
+        delete_policy(kube_apis.custom_objects, pol_name, test_namespace)
+        self.restore_default_vs(kube_apis, virtual_server_setup)
+
+    @pytest.mark.parametrize("src", [vs_policy_src])
+    def test_policy_matching_ingress_class(
+            self, kube_apis, crd_ingress_controller, virtual_server_setup, test_namespace, src,
+    ):
+        """
+        Test if policy with matching ingress class is applied to vs
+        """
+        print(f"Create rl policy")
+        pol_name = create_policy_from_yaml(kube_apis.custom_objects, policy_ingress_class_src, test_namespace)
+
+        wait_before_test()
+        policy_info = read_custom_resource(kube_apis.custom_objects, test_namespace, "policies", pol_name)
+        assert (
+                policy_info["status"]
+                and policy_info["status"]["reason"] == "AddedOrUpdated"
+                and policy_info["status"]["state"] == "Valid"
+        )
+
+        print(f"Patch vs with policy: {src}")
+        patch_virtual_server_from_yaml(
+            kube_apis.custom_objects,
+            virtual_server_setup.vs_name,
+            src,
+            virtual_server_setup.namespace,
+        )
+        wait_before_test()
+
+        vs_info = read_custom_resource(kube_apis.custom_objects, virtual_server_setup.namespace, "virtualservers", virtual_server_setup.vs_name)
+        assert (
+                vs_info["status"]
+                and vs_info["status"]["reason"] == "AddedOrUpdated"
+                and vs_info["status"]["state"] == "Valid"
+        )
+
+        delete_policy(kube_apis.custom_objects, pol_name, test_namespace)
+        self.restore_default_vs(kube_apis, virtual_server_setup)
+
+    @pytest.mark.parametrize("src", [vs_policy_src])
+    def test_policy_non_matching_ingress_class(
```

Comment:
what do you think of a negative case where VS `ingressClass` is `other` policy `ingressClass` is `nginx`, like VS won't be managed by IC but policy will?